### PR TITLE
317 - Add GA custom event for Search to DOM elements 

### PR DESCRIPTION
**Pull request by nozomione**
## Issue Number
Closing #317

## Purpose/Implementation Notes
 - Add custom events for search results to DOM elements using the GA custom events script
 - Verified that custom events are firing correctly in the `development` environment

To see the search results custom GA4 reports, please login to CCDL Analytics account, and navigate to the `development` property [`Refine.bio Web - Dev`](https://analytics.google.com/analytics/web/#/p422130981/reports/reportinghub?params=_u..nav%3Dmaui&collectionId=6630624882) (issue https://github.com/AlexsLemonade/refinebio-web/issues/310).

## Types of changes
- New feature (non-breaking change which adds functionality)

## Checklist
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by nozomione** [comment]
This PR is ready for your review, thank you David!

File: src/pages/search/index.js [hunk 1]
```diff
@@ -79,6 +80,23 @@ export const Search = ({
...
+      search
+    } = query
+
+    if (search) {
+      gtag.trackSearchTerm(userSearchTerm)
+    }
+
+    if (!!organism || !!platform || !!technology) {
+      gtag.trackFilterCombination(query)
+    }
```
Comment (davidsmejia) [inline src/pages/search/index.js:97]:
```suggestion
    gtag.trackFilterCombination(query)
```

We can wrap the two tracking calls here into one exposed `trackSearchQuery` and then conditionally track the different parts under the analytics hood.

Comment (nozomione) [reply src/pages/search/index.js]: I've renamed the methods called from `trackSearchQuery` to `trackSearchTermFromQuery` and `trackSearchTermFromQuery` respectively (pls see at f7159492683e61bc53cc65cfa75bc412a270c606).

**Comment by nozomione** [comment]
I've applied the feedback and this PR is ready for your review. Thank you David!

**Review by davidsmejia — APPROVED** [review]
An optional suggestion for naming:
`trackSearchQuery` 👍 
`trackFilterCombinationFromQuery` -> `trackSearchQueryFilterCombination`
`trackSearchTermFromQuery` -> `trackSearchQueryTerm`

**Comment by nozomione** [comment]
> An optional suggestion for naming: 
`trackFilterCombinationFromQuery` -> `trackSearchQueryFilterCombination` `trackSearchTermFromQuery` -> `trackSearchQueryTerm`

These names are great! I've applied your suggestions at a32ce44bdcab3239f9a2779d98e697b16ee0f1cf 👍 Thank you!

**Pull request merged**